CMakeLists.txt

```diff
@@ -244,7 +247,12 @@ message(STATUS "Build Type: ${CMAKE_BUILD_TYPE}")
 
 # Build with C++11 and SSE3 by default
 # TODO(wesm): These compiler warning suppressions should be removed one by one
-SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -msse3 -Wall")
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall")
+
+if (PARQUET_USE_SSE)
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse4.2")
```

Comment:
Users (me, for example) may need to build portable binaries -- my understanding is that `-march=native` can cause problems for that use case (but I don't know for sure)? My intent here was to build with SSE instructions that can be optionally disabled at runtime (using `CpuInfo`) -- this is what Impala does to generate portable x86 binaries that use SSE4 on machines that have it but not requiring separate binaries for earlier CPUs.